alf/algorithms/algorithm.py

```diff
@@ -297,11 +312,24 @@ def get_optimizer_info(self):
         optimizer_info = []
         for alg_name, opt, module_set in optimizer_and_module_sets:
             optimizer_info.append(
-                'Algorithm: "%s" Optimizer: "%s" Modules: "%s"' \
-                    % (alg_name,
-                       opt.get_config() if opt is not None else None,
-                       '; '.join(sorted([m.name for m in module_set if m is not None]))))
-        return '\n\n'.join(optimizer_info)
+                dict(
+                    Algorithm=alg_name,
+                    Optimizer=opt.get_config() if opt is not None else None,
+                    Modules=[
+                        dict(
+                            ModuleName=m.name,
+                            TrainableVariables=[
+                                v.name for v in _get_trainable_vars(m)
+                            ]) for m in module_set if m is not None
+                    ]))
+        json_pretty_str_info = json.dumps(
```

Comment:
what's the difference with pprint.pprint()? Why get_json_type?

Reply:
optimizer config contain non primitive type attrs